crates/uniswapx-rs/src/order.rs

```diff
@@ -133,6 +195,59 @@ impl ExclusiveDutchOrder {
     }
 }
 
+impl PriorityOrder {
+    pub fn decode_inner(order_hex: &[u8], validate: bool) -> Result<Self, Box<dyn Error>> {
+        Ok(PriorityOrder::decode_single(order_hex, validate)?)
+    }
+
+    pub fn encode_inner(&self) -> Vec<u8> {
+        PriorityOrder::encode_single(self)
+    }
+
+    pub fn resolve(&self, block_number: u64, timestamp: u64, priority_fee: Uint<256, 4>) -> OrderResolution {
+        let timestamp = Uint::from(timestamp);
+
+        if self.info.deadline.lt(&timestamp) {
+            return OrderResolution::Expired;
+        };
+
+        let input = self.input.scale(priority_fee);
+        let outputs = self
+            .outputs
+            .iter()
+            .map(|output| output.scale(priority_fee))
+            .collect();
+
+        if Uint::from(block_number).lt(&self.cosignerData.auctionTargetBlock.saturating_sub(Uint::from(1))) {
+            return OrderResolution::NotFillableYet;
+        };
+
+        OrderResolution::Resolved(ResolvedOrder { input, outputs })
+    }
+}
+
+impl PriorityInput {
+    pub fn scale(&self, priority_fee: Uint<256, 4>) -> ResolvedInput {
+        let amount = self.amount.wrapping_mul(Uint::from(1e7).wrapping_add(priority_fee.wrapping_mul(self.mpsPerPriorityFeeWei))).wrapping_div(Uint::from(1e7));
```

Comment:
did you run `cargo fmt`? surprised this didn't wrap if so

Reply:
ah not yet, thanks!